compiler/parser/Makefile

```diff
@@ -59,24 +53,14 @@ FORCE:
 include $(COMPILER_ROOT)/make/Makefile.compiler.foot
 
 
-# Use chapel.tab.h as prerequisite, to ensure bison only invoked once, even
+# Use bison-chapel.h as prerequisite, to ensure bison only invoked once, even
 # with parallel execution.
 #
-# no-error required here for bison 1.875 (not 1.875d)
-$(OBJ_SUBDIR)/chapel.tab.o: chapel.tab.h $(OBJ_SUBDIR_MADE) $(LEXYACC_INCLUDES)
-	$(CXX) -c $(COMP_CFLAGS) $(COMP_CFLAGS_NONCHPL) -o $@ chapel.tab.cpp $(OBJ_SUBDIR_MADE) $(LEXYACC_INCLUDES)
-
-
-# no-error required here for flex 2.5.33
-$(OBJ_SUBDIR)/lex.yy.o: lex.yy.cpp $(OBJ_SUBDIR_MADE) chapel.tab.h $(LEXYACC_INCLUDES)
-	$(CXX) -c $(COMP_CFLAGS) $(COMP_CFLAGS_NONCHPL) -o $@ $<
+$(OBJ_SUBDIR)/bison-chapel.o: ../include/bison-chapel.h bison-chapel.cpp $(OBJ_SUBDIR_MADE)
+	$(CXX) -c $(COMP_CFLAGS) $(COMP_CFLAGS_NONCHPL) -o $@ bison-chapel.cpp
 
-#
-# need to handle these ones specially as well on intel because it
-# #includes chapel.tab.h
-#
-$(OBJ_SUBDIR)/parser.o: parser.cpp $(OBJ_SUBDIR_MADE) chapel.tab.h
-	$(CXX) -c $(COMP_CFLAGS) $(COMP_CFLAGS_NONCHPL) -o $@ $<
+$(OBJ_SUBDIR)/flex-chapel.o: ../include/bison-chapel.h ../include/flex-chapel.h flex-chapel.cpp $(OBJ_SUBDIR_MADE)
+	$(CXX) -c $(COMP_CFLAGS) $(COMP_CFLAGS_NONCHPL) -o $@ flex-chapel.cpp
 
-$(OBJ_SUBDIR)/processTokens.o: processTokens.cpp $(OBJ_SUBDIR_MADE) chapel.tab.h
-	$(CXX) -c $(COMP_CFLAGS) $(COMP_CFLAGS_NONCHPL) -o $@ $<
+$(OBJ_SUBDIR)/parser.o:  ../include/bison-chapel.h ../include/flex-chapel.h parser.cpp $(OBJ_SUBDIR_MADE)
+	$(CXX) -c $(COMP_CFLAGS) $(COMP_CFLAGS_NONCHPL) -o $@ parser.cpp
```

Comment:
On Feb 17, 2015, at 2:40 PM, Lydia Duncan <notifications@github.com<mailto:notifications@github.com>> wrote:


In compiler/parser/Makefile<https://github.com/chapel-lang/chapel/pull/1331#discussion_r24862105>:

>
> -$(OBJ_SUBDIR)/processTokens.o: processTokens.cpp $(OBJ_SUBDIR_MADE) chapel.tab.h
> -     $(CXX) -c $(COMP_CFLAGS) $(COMP_CFLAGS_NONCHPL) -o $@ $<
> +$(OBJ_SUBDIR)/parser.o:  ../include/bison-chapel.h ../include/flex-chapel.h parser.cpp $(OBJ_SUBDIR_MADE)
> +     $(CXX) -c $(COMP_CFLAGS) $(COMP_CFLAGS_NONCHPL) -o $@ parser.cpp


It looks like we no longer compile processTokens.cpp here. Is that intentional? I didn't see the file get removed, is it still used?

—
Reply to this email directly or view it on GitHub<https://github.com/chapel-lang/chapel/pull/1331/files#r24862105>.

It was removed in a previous merge but this Makefile was overlooked at that time (most rules are implicit).  I happened to notice it
for this PR.

Mike